herbert_core/classes/MPIFramework/@MessagesFilebased/private/lock_.m

```diff
@@ -1,4 +1,21 @@
 function lock_(filename)
 fh = fopen(filename,'wb');
 fwrite(fh,'lk');
-fclose(fh);
\ No newline at end of file
+fclose(fh);
+
+% either smart move or delay timer to wait until FS recognises file
+present = exist(filename,'file')==2;
+n_attempts = 0;
+n_tries = 100;
+while ~present
+    pause(0.1)
+    present = exist(filename,'file')==2;
+    n_attempts = n_attempts+1;
+    if n_attempts > n_tries
+        warning('can not find write lock %s on the file system. Proceeding regardless',...
```

Comment:
if the lockfile doesn't appear in the file system. what is the risk - the reader will see the message file and no lockfile so will reads a partial data file and assume it is complete? That would silently create an error resulting combined data set

Would it be safer to fail in this situation?

Reply:
the situation this happens is, to be fair, unrealistic. The justification of the previous code, which was checking the appearance of file on the file-system was the introduction of delay mechanism, which should expire immediately after first `exist(filename,'file')==2` if it goes into while loop, somethin very strange happen with the filesystem. As something strange indeed is happening and the code hangs-up on isiscompute, I started to elaborate further. lock itself is not important for operations if they run properly, so the strangeness may be irrelevant. If something is wrong, the code will fail on the following operations, so we will know about the reason of failure due to the warning message. 

Anyway, now its proven that the problems are not related to this code at all. What should remain is the assurances that the code would not hangs on this while loop.